app/models/phone_number_opt_out.rb

```diff
@@ -0,0 +1,44 @@
+# Represents a record of a phone number that has beed opted out of SMS in AWS Pinpoint
+# AWS maintains separate opt-out lists per region, so this helps us keep track across regions
+class PhoneNumberOptOut < ApplicationRecord
+  include EncryptableAttribute
+
+  encrypted_attribute(name: :phone)
+
+  # @return [PhoneNumberOptOut, nil]
+  def self.find_by_phone(phone_number)
+    normalized = normalize(phone_number)
+
+    find_by(phone_fingerprint: [
+      Pii::Fingerprinter.fingerprint(normalized),
+      *Pii::Fingerprinter.previous_fingerprints(normalized),
+    ])
+  end
+
+  # @return [PhoneNumberOptOut]
+  def self.create_or_find_by_phone(phone_number)
+    normalized = normalize(phone_number)
+    create_or_find_by!(phone_fingerprint: Pii::Fingerprinter.fingerprint(normalized)).tap do |row|
+      if row.encrypted_phone.blank?
+        row.phone = normalized
+        row.save!
+      end
+    end
+  end
+
+  class << self
+    alias_method :mark_opted_out, :create_or_find_by_phone
+  end
+
+  def formatted_phone
+    self.class.normalize(phone)
+  end
+
+  def opt_in
+    destroy
+  end
```

Comment:
I was on the fence about deleting rows when we're done

Pros: keeps less PII in the database
Cons: might be harder to debug

Reply:
where does this: "When trying to opt-in, try to opt in across all regions" occur?

Reply:
I see opt_in_phone_number but can't see where it's called

Reply:
ah ok i see it in the controller